### PR TITLE
Fixes #43 Add oauth_body_hash

### LINKED ISSUE #43: oauth_body_hash is missing
Hi,

After sending a post request with a json payload, I see that the request was sent without oauth_body_hash in the Authorization header. Here is the code:

```
var request = require('request');
var crypto = require('crypto');
var OAuth = require('oauth-1.0a');

 var oauth = OAuth({
      consumer: {
        key: KEY,
        secret: SECERET
      },
      signature_method: 'HMAC-SHA1',
      hash_function: function (base_string, key) {
        return crypto.createHmac('sha1', key).update(base_string).digest('base64');
      }
    });

    var request_data = {
      url: http://example.com/api,
      method: 'POST',
      data: {
          key: "value"
      } 
    };

    let headers = oauth.toHeader(oauth.authorize(request_data))
    headers = Object.assign({}, headers, {
      'Content-Type': "application/json+1"
    })

    request({
      url: request_data.url,
      method: request_data.method,
      body: JSON.stringify(request_data.data),
      headers
    }, function (error, response) {
        //further code
    }); 
```

### PR DESCRIPTION
**Pull request by abutterf**

### CODE REVIEW

**Comment by ddo** [comment]
is there any service that requires **oauth_body_hash**?
we need some tests i guess

**Comment by abutterf** [comment]
Our use case is fairly complex. We need it in the context of Learning Tools Interoperability (http://www.imsglobal.org/specs/ltiv2p0/implementation-guide#toc-59). Instructure's Canvas supports this standard but testing against Canvas is not feasible because of the extra machinery required by the LTI 2.0 specification.

We have manually tested this against Instructure's Canvas and it works. I would love to be able to add some integration tests but I haven't been able to find a sevice yet that simply takes a key and secret and that supports oauth_body_hash signing.

I would be happy to add some unit tests around the functionality we added. Would that be sufficient?

**Comment by ddo** [comment]
that would be good

**Comment by ddo** [comment]
check this test https://github.com/ddo/oauth-1.0a/blob/master/test/twitter_sample_data.js
i modify some method to force it always return the same hash for testing.
you can create a new test file and apply your case

**Comment by abutterf** [comment]
@ddo I added some tests. The failed build looks like it may have been caused by transient network failures while talking to Twitter. I'm not sure how to retrigger the build.

**Comment by ddo** [comment]
https://github.com/ddo/oauth-1.0a/releases/tag/2.1.0

File: oauth-1.0a.js [hunk 1]
```diff
@@ -95,6 +99,16 @@ OAuth.prototype.getSignature = function(request, token_secret, oauth_data) {
 };
 
 /**
+ * Create a OAuth Body Hash
+ * @param {Object} request data
+ */
+OAuth.prototype.getBodyHash = function(request, token_secret) {
+  var body = typeof request.data === 'string' ? request.data : JSON.stringify(request.data)
+
+  return this.hash_function(body, this.getSigningKey(token_secret))
```
Comment (dinoboff) [inline oauth-1.0a.js]: Assuming hmac-sha1 signature, the body hash should sha1.

E.g. https://tools.ietf.org/id/draft-eaton-oauth-bodyhash-00.html#anchor15

**Pull request merged**